include/nuttx/i2c/i2c_slave.h

```diff
@@ -164,6 +208,8 @@ struct i2c_slaveops_s
         int buflen);
   int (*registercallback)(FAR struct i2c_slave_s *dev,
         int (*callback)(FAR void *arg), FAR void *arg);
+  int (*registercallback_len)(FAR struct i2c_slave_s *dev,
+        int (*callback_len)(FAR void *arg, size_t rx_len), FAR void *arg);
```

Comment:
why do we need both `registercallback` and `registercallback_len`? Maybe it is reasonable to extend `registercallback` with length parameter? Or I'm missing use cases?

Reply:
I would have preferred to update the original registercallback with a length parameter, but that would have broken any existing program based on the original.  I did not want to change existing API.

Reply:
i2c_slave.h isn't used in many place in the current code base, so it's still good time to modify the prototype. BTW, since the upgrade path is very clear, it's very easy to update the old code to fix the new interface.

Reply:
I modified the original callback to include the length and removed the second one.